상품 조회 기능 구현했습니다.

Lee-JongSoo opened the pull request:
구현 리스트입니다.

- 주문 전체 조회
- 회원 ID로 주문 조회
- 각각의 test code
- 주문 전체 조회 및 회원 ID로 주문 조회 페이징 
- 회원 전체 조회 및 회원 ID로 회원 조회 페이징 

처음 기획에서 회원 ID로 주문 조회는 완전한 id로 검색을 하는 기획이었는데  코딩을 하다보니
querydsl을 사용하여 검색어에 포함된 모든 회원ID로 주문을 검색하는게 조금 더 좋지 않을까 해서 일단 querydsl을 통한 기능으로 구현하였습니다. 다른 분들은 이에 대해 어떻게 생각하시나요??

Lee-JongSoo committed:
Feat: 주문 전체 조회 기능 구현

Lee-JongSoo committed:
Test: 회원 전체 조회 test 구현

Lee-JongSoo committed:
Feat: 회원ID로 주문 조회 기능 구현

Lee-JongSoo committed:
Test: 회원 ID로 주문 조회 검색 기능 구현

geonjunk reviewed (approved):
확인했습니다!

저는 동의합니다. 하나만 나오는 것보단 더 유용할 것 같습니다.

Lee-JongSoo committed:
Feat: 페이징 로직 추가했습니다

Lee-JongSoo committed:
Feat: 주문 전체 조회 및 회원 ID로 주문조회에 페이징 기능 추가하였습니다.

Lee-JongSoo committed:
Refactor: 주문 조회 및 회원 조회 customRepository 중복 내용 함수 추출 및 where() 조건문 함수로 추출 리펙토링하였습니다.

Lee-JongSoo committed:
Refactoring: 읽기 좋은 코드로 리펙토링하였습니다.

--리펙토링 내역--

1. AdminCocktailServiceImpl & AdminItemServiceImpl 에서 when의 잦은 사용으로 코드 읽기가 불편하여 if else문으로 리펙토링하였습니다.

2. AdminMemberRespositoryCustomImpl -> AdminMemberRespositoryImpl 로 구현체 이름 리펙토링하였습니다.

3. AdminOrderRespositoryCustomImpl -> AdminOrderRespositoryImpl 로 구현체 이름 리펙토링하였습니다.

4. DTO 이름들 변경 ex) 엔티티 to DTO, DTO to 엔티티 형식으로 리펙토링하였습니다.

5. AdminCocktailRepository의 existsByCocktailName함수의 return 타입이 String?이라 함수이름을 findCocktailNameByCocktailStatus로 리펙토링하였습니다.

KyungeunOh reviewed (approved):
넵 확인했습니다! 검색어에 포함된 모든 회원 ID로 주문을 검색하는 것 좋다고 생각합니다!

chae001122 reviewed (approved):
넵 확인했습니다. 동의합니다~~
